lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -171,10 +172,7 @@ def add_billing_address(post, payment, options)
         return unless billing = options[:billing_address]
 
         billing_address = {}
-        if payment.is_a?(CreditCard)
-          billing_address[:firstName] = payment.first_name if payment.first_name
-          billing_address[:lastName] = payment.last_name if payment.last_name
-        end
+        name_from_payment(billing_address, payment) || name_from_address(billing_address, options) if payment.is_a?(CreditCard)
```

Comment:
One other thing I just noticed - you could pass `billing` to `name_from_address` instead of the entire `options` hash, since it's already whittled down to just `options[:billing_address]`